src/shogun/regression/KernelRidgeRegression.cpp

```diff
@@ -138,18 +81,21 @@ bool CKernelRidgeRegression::train_machine(CFeatures *data)
 	}
 	ASSERT(kernel && kernel->has_features())
 
-	switch (m_train_func)
+	if (m_labels->get_num_labels() != kernel->get_num_vec_rhs())
```

Comment:
The matrix inversion method only has one check which is dependent on eigen's llt so can't be taken out of it.